libraries/search/lib/Encoding.ts

```diff
@@ -126,7 +126,7 @@ export abstract class Encoding {
   getDistance(objectiveFunction: ObjectiveFunction<Encoding>): number {
     if (this._objectives.has(objectiveFunction)) {
       if (Number.isNaN(this._objectives.get(objectiveFunction))) {
-        throw new TypeError(shouldNeverHappen("Encoding"));
+        throw new NumberIsNanError("objectiveFunction was NaN");
```

Comment:
It would be better to make these errors more specific (e.g., ObjectiveDistanceIsNanError)